plugins/emc/emc_vnx_file_

```diff
@@ -0,0 +1,516 @@
+			done
+		fi
+	done
+	if [ -z $nas_server_ok ]; then
+		echo "No active data movers!"
+		exit 1
+	fi
+exit 0
+fi
+
+for server in $nas_servers; do
+	$SSH nas_server -i $server | grep -q 'type *= nas'
+	if [ "$?" != 0 ]  ; then continue; fi
+	nas_server_ok=TRUE
+	if [ $STATSTYPE = "BASICDM" ] ; then
+member_elements=$($SSH server_stats $server -count 1 -terminationsummary no -titles never | grep '^[^[:space:]]')
+#       NUMCOL=5
+        IFS=$' ' read -ra graphs <<< "$member_elements"
+
+        echo "multigraph emc_vnx_cpu_percent"
+        echo "${server}_cpuutil.value ${graphs[1]}"
+
+        echo -e "\nmultigraph emc_vnx_network_kib"
+        echo "${server}_net_in.value ${graphs[2]}"
+        echo "${server}_net_out.value ${graphs[3]}"
+
+        echo -e "\nmultigraph emc_vnx_storage_kib"
+        echo "${server}_stor_read.value ${graphs[4]}"
+        echo "${server}_stor_write.value ${graphs[5]}"
+
+        member_elements=$($SSH server_stats $server -monitor kernel.memory -count 1 -terminationsummary no -titles never | grep '^[^[:space:]]')
+        IFS=$' ' read -ra graphs <<< "$member_elements"
+
+        echo -e "\nmultigraph emc_vnx_memory"
+        echo "${server}_total.value ${graphs[14]}"
+        echo "${server}_used.value ${graphs[15]}"
+        echo "${server}_free.value ${graphs[12]}"
+        echo "${server}_freebuffer.value ${graphs[1]}"
+        echo "${server}_encumbered.value ${graphs[8]}"
+
+        echo -e "\nmultigraph emc_vnx_filecache"
+        echo "${server}_highw_hits.value ${graphs[2]}"
+        echo "${server}_loww_hits.value ${graphs[6]}"
+        echo "${server}_w_hits.value ${graphs[7]}"
+        echo "${server}_hits.value ${graphs[4]}"
+	echo "${server}_lookups.value ${graphs[5]}"
+        
+        echo -e "\nmultigraph emc_vnx_fileresolve"
+        echo "${server}_dropped.value ${graphs[9]}"
+        echo "${server}_max.value ${graphs[10]}"
+        echo "${server}_used.value ${graphs[11]}"
+
+
+	fi
+	if [ $STATSTYPE = "NFS" ] ; then
+#nfs.v3.op data
+	member_elements=$($SSH server_stats $server -monitor nfs.v3.op -count 1 -terminationsummary no -titles never | sed -ne 's/^.*v3/v3/p')
+	NUMCOL=5
+	LINES=$(wc -l <<< "$member_elements")
+	while IFS=$'\n' read -ra graphs ; do
+		element+=( $graphs )
+	done <<< "$member_elements"
+
+	echo "multigraph vnx_emc_v3_calls_s"
+	for ((i=0; i<$((LINES)); i++ )); do
+		echo "${server}_${element[i*$NUMCOL]}".value "${element[i*$NUMCOL+1]}"
+	done
+
+	echo -e "\nmultigraph vnx_emc_v3_usec_call"
+	for ((i=0; i<$((LINES)); i++ )); do
+		echo "${server}_${element[i*$NUMCOL]}".value "${element[i*$NUMCOL+3]}"
+	done
+
+	echo -e "\nmultigraph vnx_emc_v3_op_percent"
+	for ((i=0; i<$((LINES)); i++ )); do
+		echo "${server}_${element[i*$NUMCOL]}".value "${element[i*$NUMCOL+4]}"
+	done
+
+	element=()
+
+#nfs.v4.op data
+	member_elements=$($SSH server_stats $server -monitor nfs.v4.op -count 1 -terminationsummary no -titles never | sed -ne 's/^.*v4/v4/p')
```

Comment:
Well, data is delivered in a human-readable way, like this:
`[sysadmin@mnemonic0 ~]$ server_stats server_2 -monitor nfs-std
server_2     Total        NFS        NFS     NFS Avg       NFS        NFS     NFS Avg       NFS    
Timestamp     NFS         Read      Read    Read Size     Write      Write   Write Size    Active  
             Ops/s       Ops/s      KiB/s     Bytes       Ops/s      KiB/s     Bytes      Threads  
21:44:01         1077           3        2         809        1012     2093        2118           4
21:44:16         1062           3      104       35561        1007     2203        2241           5
21:44:31         1081           3       71       23173        1021     2383        2389           4

server_2     Total        NFS        NFS     NFS Avg       NFS        NFS     NFS Avg       NFS    
Summary       NFS         Read      Read    Read Size     Write      Write   Write Size    Active  
             Ops/s       Ops/s      KiB/s     Bytes       Ops/s      KiB/s     Bytes      Threads  
Minimum          1062           3        2         809        1007     2093        2118           4
Average          1073           3       59       19848        1013     2226        2249           4
Maximum          1081           3      104       35561        1021     2383        2389           5`

They don't have any common format, and each monitoring command can do this in its own way. I would say that we use "magic string parser" to convert it into machine-readable.

Still don't have any idea about how to precisely write a comment about it